### PR TITLE
Fixed toolbar button color isn't selected properly for custom theme

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/25597

Don't overwrite chrome theme's color when custom theme is used on norma profile.

Custom theme from webstore:
image
Brave dark:
image
Brave light:
image

GTK theme on linux
image

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

`BraveThemeServiceTest.ColorProviderTest`

See the linked issue for manual testing

### CODE REVIEW

File: test/data/extensions/theme/manifest.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (sangwoo108) [inline test/data/extensions/theme/manifest.json:1]: copyright?

File: browser/themes/brave_theme_service_browsertest.cc [hunk 2]
```diff
@@ -33,7 +37,56 @@
...
+#include "extensions/browser/crx_file_info.h"
+#include "extensions/browser/extension_dialog_auto_confirm.h"
+#include "extensions/browser/extension_system.h"
+#include "extensions/browser/test_extension_registry_observer.h"
+#endif
+
+class BraveThemeServiceTest : public InProcessBrowserTest {
+ public:
+  BraveThemeServiceTest() { brave::RegisterPathProvider(); }
+
```
Comment (sangwoo108) [inline browser/themes/brave_theme_service_browsertest.cc:55]: nit: ~BraveThemeServiceTest() override = default;

File: browser/themes/brave_theme_service_browsertest.cc [hunk 3]
```diff
@@ -33,7 +37,56 @@
...
+  bool UsingCustomTheme(const ThemeService& theme_service) {
```
Comment (sangwoo108) [inline browser/themes/brave_theme_service_browsertest.cc:56]: nit: const
```suggestion
  bool UsingCustomTheme(const ThemeService& theme_service) const {
```

File: browser/themes/brave_theme_service_browsertest.cc [hunk 4]
```diff
@@ -33,7 +37,56 @@
...
+    return !theme_service.UsingSystemTheme() &&
+           !theme_service.UsingDefaultTheme();
+  }
+
+  base::FilePath GetTestDataDir() {
```
Comment (sangwoo108) [inline browser/themes/brave_theme_service_browsertest.cc:61]: This seems not to be used when `BUILDFLAG(ENABLE_EXTENSIONS)` like InstallExtension() below.

File: browser/ui/color/brave_color_mixer.cc [hunk 5]
```diff
@@ -388,13 +337,79 @@ void AddBraveDarkThemeColorMixer(ui::ColorProvider* provider,
...
+    color = SK_ColorWHITE;
+  }
+  return color_utils::BlendTowardMaxContrast(
+      color, base::ClampRound(omnibox_opacity * 0xff));
+}
+
+void AddBravifiedChromeThemeColorMixer(
+    ui::ColorProvider* provider,
+    const ui::ColorProviderManager::Key& key) {
+  if (key.custom_theme)
```
Comment (sangwoo108) [inline browser/ui/color/brave_color_mixer.cc:394]: So, this is true when custom theme is installed on normal window or it's private window right? but the latter won't call this, as it always wants dark theme.

**Review by sangwoo108 — APPROVED** [review]
LGTM with nits 👍

**Review by iefremov — APPROVED** [review]
chromium_src LGTM

File: browser/themes/brave_theme_service_browsertest.cc (hunk 2 above)
Comment (simonhong) [reply browser/themes/brave_theme_service_browsertest.cc:56]: Added.

File: browser/themes/brave_theme_service_browsertest.cc (hunk 3 above)
Comment (simonhong) [reply browser/themes/brave_theme_service_browsertest.cc]: Added.

File: browser/themes/brave_theme_service_browsertest.cc (hunk 4 above)
Comment (simonhong) [reply browser/themes/brave_theme_service_browsertest.cc:63]: Moved all new methods into extensions buildflag.

File: test/data/extensions/theme/manifest.json (hunk 1 above)
Comment (simonhong) [reply test/data/extensions/theme/manifest.json:1]: Hmm, not sure we can add comments to json file type.

File: browser/ui/color/brave_color_mixer.cc (hunk 5 above)
Comment (simonhong) [reply browser/ui/color/brave_color_mixer.cc:394]: Correct.

**Pull request merged**